docs/spec/src/integration/spec/6-secure-integration.md

```diff
@@ -1,16 +1,106 @@
+- Host: the name refers to an entity hostinh the preimage server behind the preimage oracle.
+
+![](../../assets/integration/eigenda-blob-derivation.png)
+
+### Terminal states
+All inputs to EigenDA derivation pipeline ends in exactly one of four terminal states:
+| State               | Meaning                                                             |
+| ------------------- | ------------------------------------------------------------------- |
+| **dropped**         | The input is rejected and therefore ignored by the rollup execution |
+| **stalled**         | Required data is temporarily unavailable.                           |
+| **panic**           | A fundamental invariant (e.g., field-element validity) is violated. |
+| **roll-up payload** | Success case when the desired payload bytes are produced.           |
+
+### Failure cases
+
+If an arrow in the diagram is labelled failed, the supplied data did not satisfy EigenDA’s safeguards. The DA Cert is discarded and nothing is forwarded downstream.
+
+A path marked stall means the host cannot (yet) provide the requested data.
+A path marked panic means the blob’s coefficients are not valid BN254 field elements. Then in theory, a KZG commitment cannot be generated, therefore derivation panic.
+
+- parse failed – The batcher submitted an improperly-serialized DA Cert.
+- recency check failed – The DA Cert reached the roll-up inbox too late.
+- cert validity check failed – Either the certificate does not satisfy the [quorum-attestation constraint](#2-cert-validation) or the host lies about its validity via the preimage oracle.
+- decode blob failed – The host supplies an EigenDA blob that cannot be decoded into back to rollup payload per the [spec](./3-datastructs.md#data-structs). This can occur when
+  - the batcher intentionally corrupts the encoding, or
+  - the host transmits incorrect data.
+
+But if none of these errors arise, the derivation pipeline outputs the expected payload.
+
+## EigenDA Blob derivation and Preimage Oracle
+
+We have defined the canonical derivation process, and now we described two concrete implementations and their preimage oracle:
+
+### EigenDA-proxy (Go)
+
+Eigenda proxy is the first implementation of the eigenda blob derivation described above, it is also the first implementation  of the  preimage oracle
+server that uses the external RPC to implement the preimage oracle. The proxy also contains services like http server, secondary storage and more features on [proxy github page](https://github.com/Layr-Labs/eigenda-proxy?tab=readme-ov-file#features-and-configuration-options-flagsenv-vars). Below we highlight some parts around derivation pipeline
+- the calldata input in proxy is received from http listening port (typically the requester is a roll-up consensus node).
+- proxy has both ETH RPC and EigenDA network RPC to get the preimage like the diagram below
```

Comment:
"get the preimage". You never mentioned anywhere what the "preimage" means. Like its a preimage to what? the serialized cert. Typically preimage is preimage of a hash, so this is a bit of a stretch. So we should define it at the very beginning imo